newsroom/types.py

```diff
@@ -59,12 +60,33 @@ class User(UserData):
     pass
 
 
+class UserAuth(TypedDict):
+    _id: str
+    email: str
+    password: str
+    is_enabled: bool
+    is_approved: bool
+
+
+class AuthProviderType(Enum):
+    PASSWORD = "password"
+    GOOGLE_OAUTH = "google_oauth"
+    SAML = "saml"
+
+
+class AuthProvider(TypedDict):
```

Comment:
I think it might be useful to have some notion of what does provider support and based on that we could do the checks in the auth workflow, so instead of:
```py
if auth_provider["auth_type"] == AuthProviderType.PASSWORD.value:
    send_token(...)
```
there would be something like:
```
if auth_provider.features.verify_email:
    send_token(...)
```